lib/open_food_network/scope_variants_for_search.rb

```diff
@@ -41,6 +42,14 @@ def distributor
       Enterprise.find params[:distributor_id]
     end
 
+    def apply_stock_availability_scope
+      if params[:include_out_of_stock] == "1"
+        @variants = @variants.where.not(variant_in_stock_sql)
+      else
+        @variants = @variants.where(variant_in_stock_sql)
+      end
```

Comment:
There's a couple of issues here. The second part of the conditional works, and restricts the results to variants that are in stock. The first part doesn't quite work as expected; it returns _only_ variants that are out of stock, and not variants _including_ those that are out of stock.

We can just do this instead:
```ruby
def apply_stock_availability_scope
  return if params[:include_out_of_stock] == "1"

  @variants = @variants.where(variant_in_stock_sql)
end
```

It needs a bit more detail in the tests though to cover this behaviour accurately.

Reply:
Thanks @Matt-Yorkley I have reworked this now updated the tests to cover all the possibilities _I think_.